Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGToDBAnalyzer.cs

```diff
@@ -59,103 +55,360 @@ private static void WriteXLGToDBHelp(HelpWriter writer)
+                Console.WriteLine("Processed {0} events so far.", eventCount);
+                Console.WriteLine("Total time elapsed: {0} seconds", m_stopWatch.ElapsedMilliseconds / 1000.0);
+            }
 
-        public XLGToDBAnalyzer(AnalysisInput input) : base(input) { }
+            WorkerID.Value = workerId;
+            data.Metadata.LogToTarget(data, this);
+        }
 
         /// <inheritdoc/>
         public override void Prepare()
         {
-            try
+            // If a directory exists and it contains sst files (meaning it was a previous RocksDB instance), delete the directory
+            // Also this way you don't accidentally nuke a directory that you passed in, but wasn't a previous RocksDB instance.
+            if (Directory.Exists(OutputDirPath) && Directory.GetFiles(OutputDirPath, "*.sst").Length > 0)
             {
                 Directory.Delete(path: OutputDirPath, recursive: true);
             }
-            catch (Exception e)
-            {
-                Console.WriteLine("No such dir or could not delete dir with exception {0}.\nIf dir still exists, this analyzer will append data to existing DB.", e);
-            }
 
             var accessor = KeyValueStoreAccessor.Open(storeDirectory: OutputDirPath);
 
             if (accessor.Succeeded)
             {
-                Accessor = accessor.Result;
+                m_accessor = accessor.Result;
                 m_accessorSucceeded = true;
             }
             else
             {
                 Console.Error.WriteLine("Could not access RocksDB datastore. Exiting analyzer.");
             }
+
+            m_eventCount = 0;
         }
 
         /// <inheritdoc/>
         public override int Analyze()
         {
-            if (!m_accessorSucceeded)
-            {
-                return 0;
-            }
+            m_accessor.Dispose();
+            Console.WriteLine("Num events ingested = {0}", m_eventCount);
+            Console.WriteLine("Total time elapsed: {0} seconds", m_stopWatch.ElapsedMilliseconds / 1000.0);
+            return 0;
+        }
 
-            Analysis.IgnoreResult(
-              Accessor.Use(database =>
-              {
-                  foreach (var invEvent in m_invocationEventList.BXLInvEventList)
-                  {
-                      var eq = new EventTypeQuery
-                      {
-                          EventTypeID = (int)ExecutionEventId.DominoInvocation,
-                          UUID = invEvent.UUID
-                      };
-
-                      database.Put(eq.ToByteArray(), invEvent.ToByteArray());
-                  }
-              })
-            );
+        /// <inheritdoc/>
+        public override void Dispose()
+        {
+            var ec = new Xldb.EventCount
+            {
+                Value = (uint)m_eventCount
+            };
 
-            Accessor.Dispose();
-            return 0;
+            WriteToDb(Encoding.ASCII.GetBytes("EventCount"), ec.ToByteArray());

```

Comment:
not sure that I like having this in a dispose method. can we move it to somewhere?

Reply:
Oh interesting, this should not be in the dispose method. I think some stuff got moved around. m_accessor.Dipose() should have been here.

---
In reply to: [309375984](https://github.com/microsoft/BuildXL/pull/671#discussion_r309375984) [](ancestors = 309375984)